dex-it/src/test/scala/com/wavesplatform/it/matcher/api/http/cancel/CancelOrdersByAddressAndIdsSpec.scala

```diff
@@ -29,77 +27,64 @@ class CancelOrdersByAddressAndIdsSpec extends MatcherSuiteBase with ApiKeyHeader
+      val r = validate200Json(dex1.rawApi.cancelOrdersByIdsWithKey(alice, ids))
 
       r.success should be(true)
       r.status should be("BatchCancelCompleted")
-      r.message.head should have size orders.size
+      r.message.head should have size ids.size
 
-      r.message.foreach { m =>
-        m.foreach {
-          case util.Right(HttpSuccessfulSingleCancel(_, success, status)) =>
+      r.message.head.zipWithIndex.foreach { case (m, i) =>
+        m match {
+          case util.Right(HttpSuccessfulSingleCancel(id, success, status)) =>
+            id should be(ids(i))
             success should be(true)
             status should be("OrderCanceled")
+            dex1.api.waitForOrderStatus(wavesUsdPair, id, Status.Cancelled)
           case _ => fail(s"Unexpected response $r")
         }
       }
-
-      orders.foreach(dex1.api.waitForOrderStatus(_, Status.Cancelled))
     }
 
-    "should cancel orders by public key and signature" in {
-      val orders = placeOrders()
+    "should cancel correct orders even if there are incorrect in list" in {
+      val accepted = mkOrder(alice, wavesUsdPair, BUY, 10.waves, 1.usd)
+      val filled = mkOrder(alice, wavesUsdPair, BUY, 10.waves, 2.usd)
+      val partiallyFilled = mkOrder(alice, wavesUsdPair, BUY, 8.waves, 2.usd)
+      val notFound = mkOrder(bob, wavesUsdPair, SELL, 17.waves, 2.usd)
+      val cancelled = mkOrder(alice, wavesUsdPair, BUY, 14.waves, 1.usd)
 
-      val ts = System.currentTimeMillis
-      val sign = crypto.sign(alice, alice.publicKey ++ Longs.toByteArray(ts))
+      Set(accepted, filled).foreach(placeAndAwaitAtDex(_))
+      placeAndAwaitAtDex(cancelled)
+      placeAndAwaitAtNode(notFound)
+      placeAndAwaitAtNode(partiallyFilled)
+      dex1.api.cancelOrderById(cancelled)
 
-      val r = validate200Json(dex1.rawApi.cancelOrdersByIdsWithKeyOrSignature(
-        alice.toAddress.stringRepr,
-        orders.map(_.idStr()),
-        mkHeaders(alice.publicKey, ts, sign)
-      ))
+      val uniqueIds = Seq(accepted, filled, partiallyFilled, notFound, cancelled).map(o => o.id())
 
+      val r = validate200Json(dex1.rawApi.cancelOrdersByIdsWithKey(alice, Random.shuffle(uniqueIds ++ uniqueIds)))
       r.success should be(true)
       r.status should be("BatchCancelCompleted")
-      r.message.head should have size orders.size
-
-      r.message.foreach { m =>
-        m.foreach {
-          case util.Right(HttpSuccessfulSingleCancel(_, success, status)) =>
-            success should be(true)
-            status should be("OrderCanceled")
-          case _ => fail(s"Unexpected response $r")
-        }
-      }
+      r.message.head should have size uniqueIds.size
 
+      dex1.api.waitForOrderStatus(accepted, Status.Cancelled)
+      dex1.api.waitForOrderStatus(filled, Status.Filled)
+      dex1.api.waitForOrderStatus(partiallyFilled, Status.Cancelled)
+      dex1.api.waitForOrderStatus(notFound, Status.Filled)
```

Comment:
Interesting, it should be `Status.NotFound` 🤔

Reply:
i should rename it.
"notfound" because bob is owner of this order and when we try to cancel it with alice's signature dex answers with not-found status

Reply:
fixed